tests/tripal/entities/PermissionsTest.php

```diff
@@ -0,0 +1,167 @@
+<?php
+namespace Tests\tripal\entities;
+
+use StatonLab\TripalTestSuite\DBTransaction;
+use StatonLab\TripalTestSuite\TripalTestCase;
+use Faker\Factory;
+
+class PermissionsTest extends TripalTestCase {
+  // Uncomment to auto start and rollback db transactions per test method.
+  use DBTransaction;
+
+  /**
+   * Test that our new permissions are available.
+   *
+   * @group permissions
+   */
+  public function testPermissionsAvailable() {
+
+    $permissions = module_invoke_all('permission');
+
+    // All bundle names are bio_data_##. Content types are created on install
+    // of tripal_chado and thus all sites should have them available.
+    $bundle_name = db_query('SELECT name FROM tripal_bundle limit 1')->fetchField();
+
+    // check `view [bundle name]`
+    $tripal_permissions = [
+      "view $bundle_name",
+      "create $bundle_name",
+      "edit $bundle_name",
+      "delete $bundle_name",
+    ];
+    foreach ($tripal_permissions as $permission_name) {
+      $this->assertArrayHasKey($permission_name, $permissions,
+        "Tripal permission, $permission_name, was not available.");
+    }
+  }
+
+  /**
+   * Test the permission for a given bundle.
+   *
+   * NOTE: We only test one bundle since it should be the same for all
+   * of them (done in a loop).
+   *
+   * @group permissions
+   */
+  public function testPermissionsForUser() {
+    $faker = Factory::create();
+
+    // Create organism entity for testing.
+    $bundle_name = 'bio_data_2';
```

Comment:
I'm always scared assuming bundle number.  Can we instead look it up, o reven just double check the term_id make sense and if it doesnt exit with a warning?

Reply:
Good point! I looked it up now :-)